payment/controllers.go

```diff
@@ -162,3 +162,146 @@ func CreateTransaction(service *Service) handlers.AppHandler {
 		return nil
 	})
 }
+
+// FIXME should this be consollidated with the above?
+
+// CreateAnonCardTransactionRequest includes information needed to create a anon card transaction
+type CreateAnonCardTransactionRequest struct {
+	WalletID    uuid.UUID `json:"paymentId"`
+	Transaction string    `json:"transaction"`
+	Kind        string    `json:"kind"`
+}
+
+// CreateAnonCardTransaction creates a transaction against an order
+func CreateAnonCardTransaction(service *Service) handlers.AppHandler {
+	return handlers.AppHandler(func(w http.ResponseWriter, r *http.Request) *handlers.AppError {
+		var req CreateAnonCardTransactionRequest
+		err := requestutils.ReadJSON(r.Body, &req)
+		if err != nil {
+			return handlers.WrapError(err, "Error in request body", http.StatusBadRequest)
+		}
+
+		orderID := chi.URLParam(r, "id")
+		if orderID == "" || !govalidator.IsUUIDv4(orderID) {
+			return &handlers.AppError{
+				Message: "Error validating request url parameter",
+				Code:    http.StatusBadRequest,
+				Data: map[string]interface{}{
+					"validationErrors": map[string]string{
+						"orderID": "orderID must be a uuidv4",
+					},
+				},
+			}
+		}
+		validOrderID, err := uuid.FromString(orderID)
+		if err != nil {
+			panic(err) // Should not be possible
+		}
+
+		txInfo, err := service.wallet.SubmitAnonCardTransaction(r.Context(), req.WalletID, req.Transaction)
+		if err != nil {
+			return handlers.WrapError(err, "Error submitting anon card transaction", http.StatusBadRequest)
+		}
+
+		transaction, err := service.datastore.CreateTransaction(validOrderID, txInfo.ID, txInfo.Status, txInfo.DestCurrency, "anonymous-card", txInfo.DestAmount)
+		if err != nil {
+			return handlers.WrapError(err, "Error recording anon card transaction", http.StatusInternalServerError)
```

Comment:
This is likely to be rare - but is there any way to rollback the anonymous card transaction if we fail to record to the db

Reply:
due to the idempotency key we'll just record into the db on the next client retry